packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -1,69 +1,32 @@
+  ModalHeaderTitle
 } from "@/ui";
+import SearchBar from "@/ui/search/SearchBar";
+import { TokenPickerList } from "./TokenPickerList";
+import { TokenPickerCommonTokens } from "./TokenPickerCommonTokens";
 
 export const TokenPicker = ({ isOpen, closeAction }: ModalBaseProps) => {
+  const { tokenList } = useTokenListContext();
+
   return (
-    <div>
-      <Modal isOpen={isOpen} closeAction={closeAction}>
-        <ModalHeaderTitle title="Select a token" closeAction={closeAction} />
-        <ModalContent>
-          <div className="space-y-4">
-            <input
-              type="search"
-              placeholder="search token name or paste address"
-              className="w-full p-2 bg-surface-75 rounded-xl"
-            />
-            <div className="py-3 space-y-2">
-              <p className="text-xs font-semibold text-em-low">Common tokens</p>
-              <div className="flex items-center space-x-3">
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  Eth
-                </div>
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  Pepe
-                </div>
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  usdc
-                </div>
-              </div>
-            </div>
-            <div className="relative">
-              <div className="-mx-4 overflow-y-auto border-t h-72 border-surface-50">
-                {[
-                  "1Inch",
-                  "Aave",
-                  "DAI",
-                  "USDC",
-                  "Matic",
-                  "Uni",
-                  "pepe",
-                  "Swapr",
-                  "ARB",
-                ].map((token) => (
-                  <div
-                    className="flex justify-between w-full px-4 py-2 border-b cursor-pointer border-surface-50 hover:bg-surface-50"
-                    key={token}
-                    onClick={closeAction}
-                  >
-                    <div className="flex items-center space-x-3">
-                      <div className="w-8 h-8 rounded-full bg-primary-100" />
-                      <BodyText>{token}</BodyText>
-                    </div>
-                    <BodyText>0</BodyText>
-                  </div>
-                ))}
-              </div>
-            </div>
-          </div>
-        </ModalContent>
-      </Modal>
-    </div>
+    <Modal closeAction={closeAction} isOpen={isOpen}>
+      <ModalHeaderTitle closeAction={closeAction} title="Select a token" />
+      <Divider className="my-1" />
+      <ModalContent className="mt-3 px-0">
+        <SearchBar
+          className="text-em-low mx-6"
+          placeholder="Search token name or paste address"
+        />
+        <TokenPickerCommonTokens />
+        <TokenPickerList closeAction={closeAction} tokenList={tokenList} />
```

Comment:
Talked with @Diogomartf and we think this 3 components can be declared inside this file. As they aren't being used elsewhere we think they can be declared here so we can read them were they are being used. I have a more radical opinion, not even create new components and declare all the code inside the `TokenPicker` component for a starting point. But tell us what you think :)

Reply:
OIC, I usually create different files as default but I don't mind declaring the components inside the same file if they're not being re-used.

I do have strong opinions about not splitting the code somehow, I think it's easier to spot where to make changes. If you don't mind it, I'll move the components into this file but not declare the code inside the `TokenPicker` component